scipy/integrate/_ivp/radau.py

```diff
@@ -327,6 +343,21 @@ def solve_lu(LU, b):
         self.solve_lu = solve_lu
         self.I = I
 
+        if not (mass is None):
+            if issparse(mass):
+                self.mass_matrix = csc_matrix(mass)
+                self.index_algebraic_vars = np.where(
+                  np.all(self.mass_matrix.toarray() == 0, axis=1))[0]
+            else:
+                self.mass_matrix = mass
+                self.index_algebraic_vars = np.where(np.all(
+                  self.mass_matrix == 0, axis=1))[0]
+            self.nvars_algebraic = self.index_algebraic_vars.size
```

Comment:
If there are nil rows in the mass matrix, this means that the corresponding variables are purely algebraic. Identifying these variables allows to improve the behaviour of the error estimate on the algebraic components further in `step_impl`. Indeed the original error estimation method overestimates the error on algebraic variables (see Hairer, Lubich, Roche 1989: "The Numerical Solution of Differential-Algebraic Systems by Runge-Kutta Methods"), resulting in unnecessary low time step values. Therefore in this implementation, the error on explicitly identified algebraic variables is set to 0, and the norm of the error estimate is computed only by considering the other variables.